var/spack/repos/builtin/packages/bazel/package.py

```diff
@@ -115,6 +115,7 @@ class Bazel(Package):
     patch('compile-0.6.patch',  when='@0.6:0.8')
     patch('compile-0.4.patch',  when='@0.4:0.5')
     patch('compile-0.3.patch',  when='@:0.3')
+    patch('disabledepcheck.patch')
```

Comment:
Can you explain why this patch is needed? Also, I assume it doesn't apply to every version of Bazel...

Reply:
things like this:
```
ERROR:
/tmp/spack/tf/069a40ebd843b250f49cf4908ba78fa6/external/com_google_protobuf/BUILD:107:1:
undeclared inclusion(s) in rule
'@com_google_protobuf//:protobuf_lite':

this rule is missing dependency declarations for the following files
included by
'external/com_google_protobuf/src/google/protobuf/stubs/int128.cc':

  '/dev/shm/UPSTREAM/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-7.4.0/gettext-0.20.1-7k7egp2lp56thnvdnjcrq5aihtdgavzp/include/libintl.h'
```
where this error comes from libintl.h being pulled in from <iomanip>. (there are a ton of other variants). I'm not quite sure what the rationale is for these being a fatal error rather than a warning, I'd assume the bazel folks have their reasons...

Reply:
The comment from src/main/java/com/google/devtools/build/lib/rules/cpp/HeaderDiscovery.java:

```
        // Since gcc is given only relative paths on the command line,
        // non-system include paths here should never be absolute. If they
        // are, it's probably due to a non-hermetic #include, & we should stop
        // the build with an error.
```

Reply:
In lieu of patching bazel, this also *appears* to work. I suspect that this is actually what the bazel folks were suggesting:

```
diff --git a/lib/spack/env/cc b/lib/spack/env/cc
index 08a042b..72c3f82 100755
--- a/lib/spack/env/cc
+++ b/lib/spack/env/cc
@@ -266,6 +266,7 @@ input_command="$*"
 # line. These come into the wrappers via SPACK_LDLIBS.
 #
 includes=()
+spack_includes=()
 libdirs=()
 rpaths=()
 system_includes=()
@@ -446,7 +447,7 @@ fi
 # Prepend include directories
 IFS=':' read -ra include_dirs <<< "$SPACK_INCLUDE_DIRS"
 if [[ $mode == cpp || $mode == cc || $mode == as || $mode == ccld ]]; then
-    includes=("${includes[@]}" "${include_dirs[@]}")
+    spack_includes=("${ispack_ncludes[@]}" "${include_dirs[@]}")
 fi
 
 IFS=':' read -ra rpath_dirs <<< "$SPACK_RPATH_DIRS"
@@ -501,7 +502,8 @@ args+=("${flags[@]}")
 
 # include directory search paths
 for dir in "${includes[@]}";         do args+=("-I$dir"); done
-for dir in "${system_includes[@]}";  do args+=("-I$dir"); done
+for dir in "${spack_includes[@]}";         do args+=("-isystem $dir"); done
+for dir in "${system_includes[@]}";  do args+=("-isystem $dir"); done
 
 # Library search paths
 for dir in "${libdirs[@]}";          do args+=("-L$dir"); done

```

Reply:
Very gcc-specific. I would love to hear more voices chime in here....

Reply:
Based on a few conversations in Slack, it seems this is required for a successful build of tensorflow. I'll come up with a better explanation of what's going on and open a separate PR for this.

Reply:
PR #17013 opened.